### PR TITLE
WIP: Rust binding of libstoragemgmt client API

### PR DESCRIPTION
**Pull request by cathay4t**
**Just for early review**

HowTo:
 * `sudo yum install libstoragemgmt -y`
 * `sudo systemctl start libstoragemgmt`
 * `cd rust_binding/ && cargo test`

Done:
 * All functions except LocalDisk functions.
 * All functions has unit tests.

TODO:
 * Documentation.
 * Include `cargo build` in `make`.
 * Include `cargo test` in `make check`.
 * Find a way to upload to http://crates.io/, It might need a standalone git project.

Notes:
 * No ASYNC actions any more. User should use thread instead.
 * No flags, no code is using flags yet. We may add new functions if new flags come in.

### CODE REVIEW

**Comment by tasleson** [comment]
@cathay4t Why no async support in client lib?

**Comment by tasleson** [comment]
@mulkieran FYI

File: rust_binding/src/error.rs [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+    }
+}
+
+impl From<ErrorKind> for LsmError {
+    fn from(ek: ErrorKind) -> LsmError {
+        LsmError{kind: ek}
+    }
+}
+
+pub type Result<T> = result::Result<T, LsmError>;
```
Comment (tasleson) [inline rust_binding/src/error.rs]: @cathay4t @mulkieran Is it a good idea to create the new type with the same name?  I know error chain does this, but...

Comment (cathay4t) [reply rust_binding/src/error.rs]: Error chain cannot handle 42+ errors we needed:
https://github.com/rust-lang-nursery/error-chain/issues/117

Comment (cathay4t) [reply rust_binding/src/error.rs]: Oh. Just save some typing. not a big deal.

**Comment by cathay4t** [comment]
> @cathay4t Why no async support in client lib?
In current lsmcli, there are a lot lines handling this async, if I do that in rust, I had to guide user to the
same, even for simple task as `volume_create()`. And supporting async increase the API complexity.
Comparing to thread, the only async benefit is allowing user to query progress. I am willing to sacrifice that for clear code and simple API design.

**Comment by cathay4t** [comment]
Closing to save unneeded CI works.
Will recreate PR if needed.

**Comment by cathay4t** [comment]
rust binding created in separate repo:

 * crates.io: https://crates.io/crates/libstoragemgmt-rust
 * Github: https://github.com/libstorage/libstoragemgmt-rust
 * Doc: https://libstorage.github.io/libstoragemgmt-rust

**Pull request closed without merging**